runtime/src/accounts_db.rs

```diff
@@ -4521,94 +4521,109 @@ impl AccountsDb {
+                &mut current_ancient,
+                &mut ancient_slot_pubkeys,
+                &mut dropped_roots,
             );
+        }
 
-            // could follow what shrink does more closely
-            if shrink_collect.total_starting_accounts == 0 {
-                continue; // skipping slot with no useful accounts to write
-            }
+        self.handle_dropped_roots_for_ancient(dropped_roots);
 
-            let (_, time) = measure!(current_ancient.create_if_necessary(slot, self));
-            let mut create_and_insert_store_elapsed_us = time.as_us();
-            let available_bytes = current_ancient.append_vec().accounts.remaining_bytes();
-            // split accounts in 'slot' into:
-            // 'Primary', which can fit in 'current_ancient'
-            // 'Overflow', which will have to go into a new ancient append vec at 'slot'
-            let (to_store, find_alive_elapsed) = measure!(AccountsToStore::new(
-                available_bytes,
-                &shrink_collect.alive_accounts,
-                slot
-            ));
+        total.stop();
+        self.shrink_ancient_stats
+            .total_us
+            .fetch_add(total.as_us(), Ordering::Relaxed);
 
-            ancient_slot_pubkeys.maybe_unref_accounts_already_in_ancient(
-                slot,
-                self,
-                &current_ancient,
-                &to_store,
-            );
+        self.shrink_ancient_stats.report();
+    }
 
-            let mut rewrite_elapsed = Measure::start("rewrite_elapsed");
-            // write what we can to the current ancient storage
-            let mut store_accounts_timing =
-                current_ancient.store_ancient_accounts(self, &to_store, StorageSelector::Primary);
-
-            // handle accounts from 'slot' which did not fit into the current ancient append vec
-            if to_store.has_overflow() {
-                // We need a new ancient append vec at this slot.
-                // Assert: it cannot be the case that we already had an ancient append vec at this slot and
-                // yet that ancient append vec does not have room for the accounts stored at this slot currently
-                assert_ne!(slot, current_ancient.slot());
-                let (_, time) = measure!(current_ancient.create_ancient_append_vec(slot, self));
-                create_and_insert_store_elapsed_us += time.as_us();
-
-                // write the overflow accounts to the next ancient storage
-                let timing = current_ancient.store_ancient_accounts(
-                    self,
-                    &to_store,
-                    StorageSelector::Overflow,
-                );
-                store_accounts_timing.accumulate(&timing);
-            }
-            rewrite_elapsed.stop();
+    /// put entire alive contents of 'old_storages' into the current ancient append vec or a newly created ancient append vec
+    fn combine_one_store_into_ancient(
```

Comment:
name of this function seems possibly incorrect, we could have multiple `old_storages` all being combined into the ancient, right?

Reply:
well, the type is vec, but there is only ever 1. I'm working on making the type match reality.